inst/include/vctrs.h

```diff
@@ -0,0 +1,119 @@
+#ifndef VCTRS_H
+#define VCTRS_H
+
+#include <Rinternals.h>
+#include <R_ext/Rdynload.h>
+#include <stdbool.h>
+
+static R_INLINE R_len_t vec_size(SEXP x) {
```

Comment:
It bothers me that we're not returning a long length here. It was fine internally, but for exporting we might want to return the larger type right away, otherwise we might end up with redundant functions in the API, one for each type. It is easier to directly tackle large vector than after the fact. On the other hand since vctrs primitives don't support large vectors, it wouldn't be consistent and possibly dangerous...